(GH-638) Added basic badge integration for homebrew

gep13 opened the pull request:
This relates to #638 where a request was made to support querying of homebrew package version, using the API provided by http://braumeister.org/.

I am very new to shields.io, so I may not have done this correctly.  I stitched this together by referencing a number of similar systems.

Please let me know if there is anything that needs to change.

espadrine commented on server.js:
```diff
@@ -4759,6 +4759,40 @@ cache(function(data, match, sendBadge, request) {
+      } else if (version[0] === '0') {
+        badgeData.colorscheme = 'orange';
+      } else {
+        badgeData.colorscheme = 'blue';
```
For your colors and version, I'd recommend using `var vdata = versionColor(version); badgeData.text[1] = vdata.version; badgeData.colorscheme = vdata.color;`. It ensures the version consistently starts with a `v`, and gives the same coloring scheme as the rest (which here closely matches what you have written).

espadrine commented on server.js:
```diff
@@ -4759,6 +4759,40 @@ cache(function(data, match, sendBadge, request) {
+// homebrew integration
+camp.route(/^\/homebrew\/([^\/]+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var package = match[1];
```
Could you provide an example in comment? It helps me while debugging.

espadrine commented on server.js:
```diff
@@ -4759,6 +4759,40 @@ cache(function(data, match, sendBadge, request) {
 }));
 
+// homebrew integration
+camp.route(/^\/homebrew\/([^\/]+)\.(svg|png|gif|jpg|json)$/,
```
Could you make it `/homebrew/v/PACKAGE.FORMAT` instead? The addition of the `/v/` allows more extensibility in the future, in case there are more badges made around Homebrew.

espadrine commented:
Thanks a lot for contributing!

I encourage you to add an example badge to `try.html`, so it can be presented on shields.io's front page.

gep13 commented:
@espadrine thank you for the feedback!  I won't have a chance to look at this tonight, but I will update the PR as soon as I can :+1:

zbeekman commented:
@gep13 any chance you'll be able to hack at this in the near future? I'm super stoked! If not, I can try to help you address some of @espadrine's comments.... my js is pretty nonexistent but I should be able to fumble my way through based on your work and his comments...

gep13 commented:
@zbeekman this is on my list. Hoping to get to it in the next couple days.  I have been away with work, and not had a chance to look at this.  But I haven't forgotten about it.

gep13 commented:
@espadrine please accept me apologies for the length of time it has taken to implement your suggestions.  Other things unfortunately took priority.  I think I have addressed all your suggestions, and I have rebased the changes against the current master.

Please let me know if there is anything else that needs corrected/changed.

espadrine commented on try.html:
```diff
@@ -518,6 +518,10 @@ <h3 id="version"> Version </h3>
   <td><img src='/aur/version/yaourt.svg' alt=''/></td>
   <td><code>https://img.shields.io/aur/version/yaourt.svg</code></td>
   </tr>
+  <tr><th data-keywords='homebrew'> homebrew: </th>
```
There's no need for the data-keywords here, as it is already in the text.

espadrine commented on server.js:
```diff
@@ -4814,6 +4814,36 @@ cache(function(data, match, sendBadge, request) {
+// Example: /homebrew/v/cake.svg
+camp.route(/^\/homebrew\/v\/([^\/]+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var package = match[1];
```
`package` is a reserved keyword in JS; it would be better as `pkg`.

gep13 commented:
@espadrine I was just in the process of fixing up the comments that you had made, but I see that you have now merged the PR.  Do you want another PR to correct them?

Thanks for looking at this again, really appreciate it!

espadrine commented:
> please accept me apologies for the length of time it has taken

No apology needed! Besides, going through the flux of PRs and issues makes me take a bit more than 7 days most times to get to a review.

Thanks a lot for your contribution! It's live.

espadrine commented:
> Do you want another PR to correct them?

Not necessary; I put them for exhaustivity, but I made them in a separate commit.

gep13 commented:
>@espadrine said...
Thanks a lot for your contribution! It's live.

Woot, thank you very much.  Already have it up and running:

![image](https://cloud.githubusercontent.com/assets/1271146/14520343/ebd4ef40-021b-11e6-9b47-37070b07e68e.png)
